internal/persistence/postgres/postgres.go

```diff
@@ -74,41 +83,43 @@ func (s *Postgres) Driver() persistence.Driver { return persistence.Postgres }
 
 // SetDefaultSQLOptions implements the persistence.SQLPersister interface.
 func (s *Postgres) SetDefaultSQLOptions(db *sql.DB) error {
+	// default settings are set here because the standard dbgit  client is only used for migrations.
 	db.SetMaxOpenConns(persistence.DefaultMaxConn)
 	db.SetMaxIdleConns(persistence.DefaultMaxIdleConn)
 	db.SetConnMaxLifetime(persistence.DefaultMaxConnLifetime)
 	return nil
 }
 
 func (s *Postgres) Get(ctx context.Context, key string) ([]byte, error) {
-	q := postgresQuery{query: s.readOnlyDB, queryTimeout: s.queryTimeout}
+	q := postgresQuery{query: s.readOnlyDBPool, queryTimeout: s.queryTimeout}
 	return q.Get(ctx, key)
 }
 
 func (s *Postgres) Put(ctx context.Context, key string, value []byte) error {
-	q := postgresQuery{query: s.db, queryTimeout: s.queryTimeout}
+	q := postgresQuery{query: s.dbPool, queryTimeout: s.queryTimeout}
 	return q.Put(ctx, key, value)
 }
 
 func (s *Postgres) Delete(ctx context.Context, key string) error {
-	q := postgresQuery{query: s.db, queryTimeout: s.queryTimeout}
+	q := postgresQuery{query: s.dbPool, queryTimeout: s.queryTimeout}
 	return q.Delete(ctx, key)
 }
 
 func (s *Postgres) List(ctx context.Context, prefix string, opts *persistence.ListOpts) (persistence.ListResult,
 	error,
 ) {
-	q := postgresQuery{query: s.readOnlyDB, queryTimeout: s.queryTimeout}
+	q := postgresQuery{query: s.readOnlyDBPool, queryTimeout: s.queryTimeout}
 	return q.List(ctx, prefix, opts)
 }
 
 func (s *Postgres) Tx(ctx context.Context) (persistence.Tx, error) {
-	tx, err := s.db.BeginTx(ctx, nil)
+	tx, err := s.dbPool.BeginTx(ctx, pgx.TxOptions{})
```

Comment:
With `database/sql`, a transaction will be automatically roll backed on context cancellation. This is not something we have by default with `pgx`. Although we don't have any test that is testing this scenario, I think it would be a good to do the same thing here, wdyt?

Reply:
ACK: thanks!